File: python/plugins/solaceMessageProcessor/message_builder.py
from Trade_pb2 import Trade, NewOrder, Cancel, Amend


class MessageBuilder:
    def build_new_order(self, order_id: str, symbol: str, quantity: int, price: float) -> bytes:
        trade = Trade()
        new_order = NewOrder(
            order_id=order_id,
            symbol=symbol,
            quantity=quantity,
            price=price
        )
        trade.new_order.CopyFrom(new_order)
        return trade.SerializeToString()

    def build_cancel(self, order_id: str) -> bytes:
        trade = Trade()
        cancel = Cancel(order_id=order_id)
        trade.cancel.CopyFrom(cancel)
        return trade.SerializeToString()

    def build_amend(self, order_id: str, new_quantity: int, new_price: float) -> bytes:
        trade = Trade()
        amend = Amend(
            order_id=order_id,
            new_quantity=new_quantity,
            new_price=new_price
        )
        trade.amend.CopyFrom(amend)
        return trade.SerializeToString()

File: java/plugins/solaceMessageProcessor/src/main/java/com/applicationsx/trademessageprocessor/MessageBuilder.java
package com.applicationsx.trademessageprocessor;

import fixprocessor.TradeOuterClass.Trade;
import fixprocessor.TradeOuterClass.NewOrder;
import fixprocessor.TradeOuterClass.Cancel;
import fixprocessor.TradeOuterClass.Amend;

public class MessageBuilder {

    public byte[] buildNewOrder(String orderId, String symbol, int quantity, double price) {
        NewOrder newOrder = NewOrder.newBuilder().setOrderId(orderId).setSymbol(symbol).setQuantity(quantity).setPrice(price).build();

        Trade trade = Trade.newBuilder().setNewOrder(newOrder).build();
        return trade.toByteArray();
    }

    public byte[] buildCancel(String orderId) {
        Cancel cancel = Cancel.newBuilder().setOrderId(orderId).build();
        Trade trade = Trade.newBuilder().setCancel(cancel).build();
        return trade.toByteArray();
    }

    public byte[] buildAmend(String orderId, int newQty, double newPrice) {
        Amend amend = Amend.newBuilder().setOrderId(orderId).setNewQuantity(newQty).setNewPrice(newPrice).build();

        Trade trade = Trade.newBuilder().setAmend(amend).build();
        return trade.toByteArray();
    }
}
